opennms-services/src/main/java/org/opennms/netmgt/scriptd/Executor.java

```diff
@@ -61,173 +63,119 @@
+     * The configured scripts (UEI specified).
      */
-    private final String m_name;
+    private final Map<String,List<EventScript>> m_eventScriptMap = new ConcurrentHashMap<>();
 
     /**
-     * The status of this fiber.
+     * The DAO object for fetching nodes
      */
-    private int m_status;
+    private final NodeDao m_nodeDao;
 
     /**
-     * The configuration.
+     * The BSF manager
      */
-    private ScriptdConfigFactory m_config;
+    private BSFManager m_scriptManager = null;
 
     /**
-     * The configured scripts (no UEI specified).
+     * The {@link ExecutorService} that will execute tasks for each
+     * event.
      */
-    private List<EventScript> m_eventScripts;
+    private ExecutorService m_executorService;
 
     /**
-     * The configured scripts (UEI specified).
+     * The broadcast event receiver.
      */
-    private Map<String,List<EventScript>> m_eventScriptMap;
+    private BroadcastEventProcessor m_broadcastEventProcessor;
 
     /**
-     * The BSF manager
-     */
-    private BSFManager m_mgr;
-
-    /**
-     * The DAO object for fetching nodes
+     * The configuration.
      */
-    private NodeDao m_nodeDao;
+    private ScriptdConfigFactory m_config;
 
     /**
-     * Constructs a new action daemon execution environment. The constructor
-     * takes three arguments that define the source of commands to be executed and
-     * the maximum time that a command may run.
-     * 
-     * @param execQ
-     *            The execution queue
      * @param config
      *            The <em>Scriptd</em> configuration.
      * @param nodeDao
      *            The <em>DAO</em> for fetching node information
      */
-    Executor(BlockingQueue<Event> execQ, ScriptdConfigFactory config, NodeDao nodeDao) {
-        m_execQ = execQ;
-        m_config = config;
+    Executor(ScriptdConfigFactory config, NodeDao nodeDao) {
 
-        loadConfig();
+        m_config = config;
 
-        m_worker = null;
-        m_name = "Scriptd-Executor";
-        m_mgr = null;
-        m_status = START_PENDING;
+        m_nodeDao = nodeDao;
 
-	m_nodeDao = nodeDao;
+        loadConfig();
     }
 
     /**
      * Load the m_scripts and m_scriptMap data structures from the
      * configuration.
+     * 
+     * TODO: This doesn't deregister scripts from m_eventScripts and
+     * m_eventScriptMap during config reloads.
      */
     private void loadConfig() {
 
-        EventScript[] scripts = m_config.getEventScripts();
-
-        m_eventScripts = new ArrayList<EventScript>();
-        m_eventScriptMap = new ConcurrentHashMap<String,List<EventScript>>();
-
-        for (int i = 0; i < scripts.length; i++) {
-            Uei[] ueis = scripts[i].getUei();
+        for (EventScript script : m_config.getEventScripts()) {
+            Uei[] ueis = script.getUei();
 
             if (ueis.length == 0) {
-                m_eventScripts.add(scripts[i]);
+                m_eventScripts.add(script);
```

Comment:
Is there any way loadConfig() could be called multiple times?  Since it's a COW list, could we end up with duplicates if we reload configs?